plugins/schedule/base/lib/api/KalturaScheduleEvent.php

```diff
@@ -307,6 +342,25 @@ public function validateForInsert($propertiesToSkip = array())
 		parent::validateForInsert($propertiesToSkip);
 	}
 	
+	protected function setTimingFromLinkedToEvent($linkedToEventId, $duration = null)
+	{
+		if (!is_null($linkedToEventId)) //if it is null we are unlinking
+		{

```

Comment:
if linkedToEventId will be null the function will fail in exception - remove this validation

Reply:
No it will not fail with exception. if $linkedTo is not null but $linkedToEventId is null, it means that it is a property being updated and the event is being unlinked. In this case, we do not wish to update startDate and endDate, only remove the references to the preceding event.